### PR TITLE
Update to Mongo 3.6 drivers

### PR DESCRIPTION
**Pull request by CedricPerotto**
Merged From https://github.com/tidepool-org/shoreline/pull/68

### CODE REVIEW

File: user/hasher_test.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func TestGenerateUniqueHashLength(t *testing.T) {
 
 	if theHash, err := generateUniqueHash([]string{"one", "two", "miss a few", "99", "100"}, 20); err == nil {
 		if len(theHash) != 20 {
-			t.Fatalf("the hash should be 20 characters in length but instead is %d", len(theHash))
+			t.Fatalf("the has should be 20 characters in length, but has %v", len(theHash))
```
Comment (ydrisR) [inline user/hasher_test.go]: Why removing the "h" from the Hash?

**Review by ydrisR — APPROVED** [review]
Looks good overall, just found a typo and one big question:
Why do we keep dependencies within our repo? Why not use "go get" to get everything at build time? traceability?

File: user/hasher_test.go (hunk 1 above)
Comment (CedricPerotto) [reply user/hasher_test.go]: This was in the commit from tidepool. Didn't check this.

**Comment by CedricPerotto** [comment]
> Looks good overall, just found a typo and one big question:
> Why do we keep dependencies within our repo? Why not use "go get" to get everything at build time? traceability?

"go get" doesn't allow us to get a specific release or a forked branch of a repository (which is the case for us with the go-common dependency). 
Tidepool uses dep (https://golang.github.io/dep/) which allow us to define a description file for dependencies in Gopkg.toml and Gopkg.lock, and install it through cli commands.

**Pull request merged**